### PR TITLE
Trading Modifications

### PR DESCRIPTION
**Pull request by brianweir-da**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,5 @@ yarn-debug.log*
 yarn-error.log*
 yarn.lock
 daml.js/
+
+.idea/
```
Comment (lucianojoublanc-da) [inline .gitignore:20]: Probably worth adding to your local .gitignore

File: daml/Marketplace/Trading.daml [hunk 2]
```diff
@@ -153,44 +109,93 @@ template Service
     maintainer key._1
 
     controller customer can
-      nonconsuming RequestCreateOrder : (ContractId CreateOrderRequest, ContractId AssetDeposit)
+      nonconsuming RequestCreateOrder : (ContractId Order, ContractId CreateOrderRequest, ContractId AssetDeposit)
```
Comment (lucianojoublanc-da) [inline daml/Marketplace/Trading.daml:112]: So I had split this into `Order` and `Trade` files, but @georg-da nuked them. I think that would make it tidier so we don't have to write `OrderDetails`, `OrderRequest`, `OrderStatus` etc. But maybe he was just rushed. Let's wait til he gets back to see what he thinks of that.

File: exberry_adapter/bot/exberry_adapter_bot.py [hunk 3]
```diff
@@ -4,13 +4,17 @@
...
-def get_sid() -> int:
-    global SID
-    SID = SID + 1
-    return SID
+# SID = 1 # default SID, use ExberrySID contract to change while running
+# SID = int(time.mktime(datetime.now().timetuple()))
+# def get_sid() -> int:
+#     global SID
+#     SID = SID + 1
+#     return SID
```
Comment (lucianojoublanc-da) [inline exberry_adapter/bot/exberry_adapter_bot.py:17]: delete

**Review by lucianojoublanc-da — APPROVED** [review]

File: daml/Marketplace/Trading.daml (hunk 2 above)
Comment (brianweir-da) [reply daml/Marketplace/Trading.daml:112]: This is likely due to cyclic dependency hell - I tried to split them up but found its that time consuming that I gave up and left it in its current form. I'd be happy to tackle it again later once the trading features are at a more mature stage.

File: exberry_adapter/bot/exberry_adapter_bot.py (hunk 3 above)
Comment (brianweir-da) [reply exberry_adapter/bot/exberry_adapter_bot.py:17]: 👍

**Pull request merged**